uber/models.py

```diff
@@ -165,9 +166,13 @@ class MagModel:
 
     required = ()
 
-    @property
-    def _class_attrs(self):
-        return {name: getattr(self.__class__, name) for name in dir(self.__class__)}
+    @cached_classproperty
+    def _class_attr_names(cls):
```

Comment:
I turned this into a class property instead of an instance property because it's calculated entirely from attributes in the class, not the instance. Also, I am using a `@cached_classproperty`, because the return value shouldn't change over the life of the class (technically the class attrs will be updated by Mixin classes, but always before this is called for the first time, so caching the result is still appropriate).